### ISSUE TITLE
Documentation discussion

### ISSUE THREAD

**Issue by devanshj** [cassiozen/useStateMachine#71]
@cassiozen How and where would you want me to write that document covering typescript features? Like maybe in a "Features" section? Or like in a changelog? (that would be lame haha)

And if I'm going to use shiki-twoslash we'd have to host it somewhere. So should we make a documentation website? Or if all this is on low priority I can simply add a wiki page mentioning typescript features with screenshots?

I'm not sure how to go about this haha

**Comment by cassiozen** [comment]
We will need a site at some point, given that GitHub's wiki solution is so poor, but that's not something I could take on immediately. 

Do you think you could, for now, use the wiki with screenshots and links to the TypeScript playground?

**Comment by cassiozen** [comment]
To be even more specific, I think it would be great to have a Primer, a tl;dr on the readme plus the full article on its own documentation / wiki page.

**Comment by devanshj** [comment]
> use the wiki with screenshots and links to the TypeScript playground?

Yep that's not bad, doable

> a Primer, a tl;dr on the readme

Yeah my thoughts are similar it's good to point out key-selling features on the readme and then dive into all features in details in a separate place like a wiki page

**Comment by cassiozen** [comment]
As discussed we now have a docusaurus setup under the `website` folder, with shiki-twoslash configured.

This is not hosted yet, but you can start working there and we will publish it before the final 1.0.0 release

**Comment by cassiozen** [comment]
Deployed: https://usestatemachine.js.org/

**Comment by cassiozen** [comment]
Yeah, I just copied the current readme. Send the PR and I work on properly splitting and updating the docs.

**Comment by cassiozen** [comment]
With that the only thing missing for the 1.0 release will be your article and my video, correct? Am I missing something?

We can target sendT for 1.1 and hierarchical for 1.2

**Comment by devanshj** [comment]
> Am I missing something?

Probably not, just types-lite but we can have them for later release

> We can target sendT for 1.1 and hierarchical for 1.2

Sure
